niceman/resource/session.py

```diff
@@ -385,9 +397,38 @@ def isdir(self, path):
             self.execute_command(['test', '-d', path])
             return True
         except CommandError:
+            lgr.error(cmd.msg)
+            return False
+
+    def chmod(self, mode, remote_path, recursive=False):
+        """Set the mode of a remote path
+        """
+        try:
+            command = ['chmod']
+            if recursive: command += ["-R"]
+            command += [mode] + [remote_path]
+            self.execute_command(command)
+            return True
+        except CommandError as cmd:
+            lgr.error(cmd.msg)
+            return False
```

Comment:
Why catch errors? It isn't  a `issomething` function, so I'm hoping should throw exception if something goes wrong

Reply:
I did that thinking that the command would return a success or fail status. We can throw an error instead. That is more in keeping with the control/flow of niceman.

Reply:
Well, we do throw CommandError if e.g. execution fails, so I think we should be consistent with that -- should make coding using this API easier (instead of constant checks for every command).  May be derive a new exception, e.g. `FileManipulationError`  and `TransferError` both derived from `CommandError` to be used in chmod/chown and put/get accordingly.  This way we could run sequence of commands catching the error once if needed